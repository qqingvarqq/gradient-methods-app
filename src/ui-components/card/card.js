import React from 'react';
import './card.css';

export default function Card(props) {
  const {children} = props;
  return <div className="card">{children}</div>;
}
