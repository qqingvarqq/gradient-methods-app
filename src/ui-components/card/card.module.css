.root {
  height: fit-content;
  width: 100%;
  min-height: 10px;
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 20px;
  border-radius: 8px;
  box-shadow: 0px 0px 2px 1px rgba(0, 0, 0, 0.3);
  transition: box-shadow 0.3s ease;
  cursor: default;
}
.root:hover {
  box-shadow: 0px 0px 4px 2px rgba(0, 0, 0, 0.3);
}
