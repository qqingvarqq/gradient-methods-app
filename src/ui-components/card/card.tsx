import * as React from 'react';
import styles from './card.module.css';

export const Card: React.FC<React.DetailedHTMLProps<
  React.HTMLAttributes<HTMLDivElement>,
  HTMLDivElement
>> = props => {
  const {className = '', ...rest} = props;
  return <div className={`${styles.root} ${className}`} {...rest}></div>;
};
