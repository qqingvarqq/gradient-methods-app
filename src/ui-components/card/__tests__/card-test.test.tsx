import * as React from 'react';
import {render} from '@testing-library/react';
import {Card} from '../card';

describe('Card', () => {
  it('should render without crashing', () => {
    const {asFragment} = render(<Card />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should render children', () => {
    const ChildElement = () => <div>Hello From Card</div>;
    const {asFragment} = render(
      <Card>
        <ChildElement />
      </Card>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
