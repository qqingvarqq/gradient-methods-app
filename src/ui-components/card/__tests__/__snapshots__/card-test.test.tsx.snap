// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render children 1`] = `
<DocumentFragment>
  <div
    class="root "
  >
    <div>
      Hello From Card
    </div>
  </div>
</DocumentFragment>
`;

exports[`Card should render without crashing 1`] = `
<DocumentFragment>
  <div
    class="root "
  />
</DocumentFragment>
`;
